src/pages/settings/SettingForm.tsx

```diff
@@ -19,33 +23,51 @@ export const getConfigId = (key: string) => {
 interface Props {
   configField: ConfigField;
   value?: string;
+  clusteredValue?: ClusterSpecificValues;
   isLast?: boolean;
 }
 
-const SettingForm: FC<Props> = ({ configField, value, isLast }) => {
+const SettingForm: FC<Props> = ({
+  configField,
+  value,
+  clusteredValue,
+  isLast,
+}) => {
   const { isRestricted } = useAuth();
   const [isEditMode, setEditMode] = useState(false);
   const notify = useNotify();
   const toastNotify = useToastNotification();
   const queryClient = useQueryClient();
   const { canEditServerConfiguration } = useServerEntitlements();
+  const { data: settings } = useSettings();
+  const isClustered = isClusteredServer(settings);
 
   const editRef = useRef<HTMLDivElement | null>(null);
 
   // Special cases
   const isTrustPassword = configField.key === "core.trust_password";
   const isLokiAuthPassword = configField.key === "loki.auth.password";
+  const isMaasMachine = configField.key === "maas.machine";
   const isSecret = isTrustPassword || isLokiAuthPassword;
+  const isAddress = configField.key.endsWith("_address");
+  const isVolume = configField.key.endsWith("_volume");
+  const isClusteredInput =
+    isClustered && (isMaasMachine || isAddress || isVolume);
 
   const settingLabel = (
     <ResourceLabel bold type="setting" value={configField.key} />
   );
 
-  const onSubmit = (newValue: string | boolean) => {
-    const config = {
-      [configField.key]: String(newValue),
-    };
-    updateSettings(config)
+  const onSubmit = (newValue: string | boolean | ClusterSpecificValues) => {
+    const mutationPromise =
```

Comment:
```suggestion
    const mutation =
```
IMHO a cleaner name. Though we are not very consistent in other places with naming those.